src/14-curated-discovery/01-artworks.ts

```diff
@@ -0,0 +1,286 @@
+        },
+      },
+      {
+        name: "listPriceCurrency",
+        dataType: ["text"],
+        moduleConfig: {
+          "text2vec-openai": {
+            skip: true,
+          },
+        },
+      },
+      {
+        name: "artworkLocation",
+        dataType: ["text"],
+        moduleConfig: {
+          "text2vec-openai": {
+            skip: false,
+            vectorizePropertyName: true,
+          },
+        },
+      },
+      {
+        name: "categories",
+        dataType: ["text[]"],
+        moduleConfig: {
+          "text2vec-openai": {
+            skip: false,
+            vectorizePropertyName: true,
+          },
+        },
+      },
+      {
+        name: "tags",
+        dataType: ["text[]"],
+        moduleConfig: {
+          "text2vec-openai": {
+            skip: false,
+            vectorizePropertyName: true,
+          },
+        },
+      },
+      {
+        name: "additionalInformation",
+        dataType: ["text"],
+        moduleConfig: {
+          "text2vec-openai": {
+            skip: false,
+            vectorizePropertyName: true,
+          },
+        },
+      },
+      {
+        name: "imageUrl",
+        dataType: ["text"],
+        moduleConfig: {
+          "text2vec-openai": {
+            skip: true,
+          },
+        },
+      },
+      {
+        name: "imageDescription",
+        dataType: ["text"],
+        moduleConfig: {
+          "text2vec-openai": {
+            skip: false,
+            vectorizePropertyName: true,
+          },
+        },
+      },
+      {
+        name: "artistID",
+        dataType: ["text"],
+        moduleConfig: {
+          "text2vec-openai": {
+            skip: true,
+          },
+        },
+      },
+      {
+        name: "partnerID",
+        dataType: ["text"],
+        moduleConfig: {
+          "text2vec-openai": {
+            skip: true,
+          },
+        },
+      },
+    ],
+  }
+
+  const classResult = await client.schema
+    .classCreator()
+    .withClass(classWithProps)
+    .do()
+
+  console.log(JSON.stringify(classResult, null, 2))
+}
+
+/**
+ * Insert artworks into Weaviate
+ *
+ * Also assigns a deterministic UUID to each artwork,
+ * based on its Gravity ID
+ */
+async function insertArtworks(
+  artworks: GravityArtwork[],
+  batchSize: number = BATCH_SIZE
+) {
+  console.log(`Inserting artwork: ${artworks.length}`)
+
+  const batches = _.chunk(artworks, batchSize)
+  console.log(`Inserting ${batches.length} batches`)
+
+  for (const artworkBatch of batches) {
+    let batcher = client.batch.objectsBatcher()
+    batcher = batcher.withObjects(
+      ...artworkBatch.map((artwork) => {
+        return {
+          class: CLASS_NAME,
+          properties: {
+            internalID: artwork.id,
+            slug: artwork.slug,
+            title: artwork.title,
+            date: artwork.date,
+            rarity: artwork.rarity,
+            medium: artwork.medium,
+            materials: artwork.materials,
```

Comment:
Note that we opted to use user-facing terminology for many of these fields (attribution_class → **rarity**; category → **medium**; medium → **materials**; genes → **categories**), on the assumption that this would perform better with the embedding models.